projects/plugins/jetpack/_inc/lib/class-jetpack-ai-helper.php

```diff
@@ -95,16 +102,39 @@ public static function transient_name_for_completion() {
 		return 'jetpack_openai_completion_' . get_current_user_id(); // Cache for each user, so that other users dont get weird cached version from somebody else.
 	}
 
+	/**
+	 * Mark the edited post as "touched" by AI stuff.
+	 *
+	 * @param  int $post_id Post ID for which the content is being generated.
+	 * @return void
+	 */
+	private static function mark_post_as_ai_assisted( $post_id ) {
+		if ( ! $post_id ) {
+			return;
+		}
+		$previous = get_post_meta( $post_id, self::$post_meta_with_ai_generation_number, true );
+		if ( ! $previous ) {
+			$previous = 0;
+		} elseif ( ! is_numeric( $previous ) ) {
+			// Data corrupted, nothing to do.
+			return;
+		}
+		$new_value = intval( $previous ) + 1;
+		update_post_meta( $post_id, self::$post_meta_with_ai_generation_number, $new_value );
+	}
+
 	/**
 	 * Get text back from WordPress.com based off a starting text.
 	 *
 	 * @param  string $content The content that's already been typed in the block.
+	 * @param  int    $post_id Post ID for which the content is being generated.
 	 * @return mixed
 	 */
-	public static function get_gpt_completion( $content ) {
+	public static function get_gpt_completion( $content, $post_id ) {
 		$content = wp_strip_all_tags( $content );
 		$cache   = get_transient( self::transient_name_for_completion() );
 		if ( $cache ) {
+			self::mark_post_as_ai_assisted( $post_id );
```

Comment:
I'm not sure I understand this part. If we're serving data we've already fetched (and thus incremented) before, why should we increment the counter again?

Reply:
The cache is not post-specific. If you insert an image with a dog in post 1 and later use the same prompt to generate image in post 2, but hit cache, you also use AI content in post 2

Reply:
> Since we only cache the prompt for a minute though, it's unlikely to happen, correct?

Ok, that is valid. We can remove it in the paragraph one.

Reply:
Removed